layerindex/tools/import_layers.py

```diff
@@ -0,0 +1,494 @@
+
+            layer_idmap = {}
+            exclude_fields = ['id', 'updated']
+            for layerjs in jsdata:
+                if layer_re and not layer_re.match(layerjs['name']):
+                    #logger.debug('Skipping layer %s, does not match layer restriction' % layerjs['name'])
+                    continue
+
+                layeritem = LayerItem.objects.filter(name=layerjs['name']).first()
+                if layeritem:
+                    # Already have this layer
+                    if layerjs['updated'] <= layeritem.updated and not args.reload:
+                        logger.debug('Skipping layer %s, already up-to-date' % layerjs['name'])
+                        layer_idmap[layerjs['id']] = layeritem
+                        continue
+                    else:
+                        logger.debug('Updating layer %s' % layerjs['name'])
+                else:
+                    logger.debug('Adding layer %s' % layerjs['name'])
+                    layeritem = LayerItem()
+                for key, value in layerjs.items():
+                    if key in exclude_fields:
+                        continue
+                    setattr(layeritem, key, value)
+                layeritem.save()
+                layer_idmap[layerjs['id']] = layeritem
+
+            # Get layer branches
+            logger.debug('Importing layer branches')
+            rq = urllib.request.Request(layerbranches_url)
+            data = urllib.request.urlopen(rq).read()
+            jsdata = json.loads(data.decode('utf-8'), object_hook=datetime_hook)
+
+            layerbranch_idmap = {}
+
+            def import_child_items(parentobj, objclass, childlist=None, url=None, parent_orig_id=None, parentfield=None, exclude_fields=None, key_fields=None, custom_fields=None, custom_field_cb=None):
+                logger.debug('Importing %s for %s' % (objclass._meta.verbose_name_plural, parentobj))
+
+                if parentfield is None:
+                    parentfield = parentobj.__class__.__name__.lower()
+
+                if exclude_fields is None:
+                    exclude = ['id', parentfield]
+                else:
+                    exclude = exclude_fields[:]
+                if custom_fields is not None:
+                    exclude += custom_fields
+                if key_fields is None:
+                    keys = None
+                else:
+                    # The parent field always needs to be part of the keys
+                    keys = key_fields + [parentfield]
+
+                def fetch_api_url(api_url):
+                    rq = urllib.request.Request(api_url)
+                    data = urllib.request.urlopen(rq).read()
+                    return json.loads(data.decode('utf-8'))
+
+                if url:
+                    if parent_orig_id is None:
+                        raise Exception('import_child_items: if url is specified then parent_orig_id must also be specified')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.



Similar issue at line number 203.